robot/EDA/resources/CoursesSettingsPageObject.py

```diff
@@ -20,3 +20,17 @@ def _is_current_page(self):
             locator_tab,
             error=f"Courses tab with locator '{locator_tab}' is not available on the page"
         )
+
+    def verify_text_appears(self, is_displayed):
+        """ Verify the text message is displayed
+            this message gets displayed when the 'Run copy' button is clicked
+            in both read and edit mode
+        """
+        locator_displayed = eda_lex_locators["eda_settings_courses"]["text_message_displayed"]
+        locator_hidden = eda_lex_locators["eda_settings_courses"]["text_message_hidden"]
+        if str(is_displayed).lower() == "true":
+            self.selenium.wait_until_page_contains_element(locator_displayed,
+                                                           error="Run copy text is not displayed")
+        else:
+            self.selenium.wait_until_page_contains_element(
+                locator_hidden, error="Run copy text message is displayed")
```

Comment:
may be u can do get attribute of class and verify is_hide is there instead of using 2 locators. and pass text as either variable or hardcode in the keyword instead of hardcoding in loctor

Reply:
Thanks for the suggestion and I have updated the logic for the same. Please take a look now.